test/app-tap/datetime.test.lua

```diff
@@ -1285,6 +1285,43 @@ test:test("Parse tiny date into seconds and other parts", function(test)
     test:is(tiny.timestamp, 30.528, "timestamp")
 end)
 
+test:test("Parse strptime format", function(test)
+    test:plan(16)
+    local formats = {
+        {'Thu Jan  1 03:00:00 1970',    '%c',       '1970-01-01T03:00:00Z'},
+        {'01/01/70',                    '%D',       '1970-01-01T00:00:00Z'},
+        {'01/01/70',                    '%m/%d/%y', '1970-01-01T00:00:00Z' },
+        {'Thu Jan  1 03:00:00 1970',    '%Ec',      '1970-01-01T03:00:00Z' },
+        {'1970-01-01',                  '%F',       '1970-01-01T00:00:00Z' },
+        {'1970-01-01',                  '%Y-%m-%d', '1970-01-01T00:00:00Z' },
+        -- {' 1-Jan-1970',                 '%v' },
+        {' 1-Jan-1970',                 '%e-%b-%Y', '1970-01-01T00:00:00Z' },
+        {'01/01/70',                    '%x',       '1970-01-01T00:00:00Z' },
+        {'1970-01-01T0300+0300',        '%Y-%m-%dT%H%M%z',
+            '1970-01-01T03:00:00+0300' },
+        {'1970-01-01T03:00:00+0300',    '%Y-%m-%dT%H:%M:%S%z',
+            '1970-01-01T03:00:00+0300' },
+        {'1970-01-01T03:00:00.125000000+0300',  '%Y-%m-%dT%H:%M:%S.%f%z',
+            '1970-01-01T03:00:00.125+0300' },
+        {'1970-01-01T03:00:00.125+0300',        '%Y-%m-%dT%H:%M:%S.%f%z',
+            '1970-01-01T03:00:00.125+0300' },
+        {'1970-01-01T03:00:00.125',             '%Y-%m-%dT%H:%M:%S.%f',
+            '1970-01-01T03:00:00.125Z' },
+        {'1970-01-01T03:00:00.125',             '%FT%T.%f',
+            '1970-01-01T03:00:00.125Z' },
+        {'1970-01-01T03:00:00.125+0300',        '%FT%T.%f%z',
+            '1970-01-01T03:00:00.125+0300' },
+        {'1970-01-01T03:00:00.125000000+0300',  '%FT%T.%f%z',
+            '1970-01-01T03:00:00.125+0300' },
+    }
```

Comment:
Please add test cases with minimal and max values of every part of datetime value.

Reply:
Now, after negative years support added to both parser (default, and strptime-like) I've added some tests which checks values of various date subcomponents (i.e. day in the month, hours, minutes) and also checking whole range of years supported [-5879610 .. 5879611], not only originally supported range of years [0001..9999]